Client/Frontend/Autofill/CreditCard/CreditCardBottomSheet/SingleCreditCardView.swift

```diff
@@ -0,0 +1,306 @@
+        setupView()
+    }
+
+    deinit {
+        notificationCenter.removeObserver(self)
+    }
+
+    // MARK: View Setup
+    func addSubviews() {
+        let isIpad = UIDevice.current.userInterfaceIdiom == .pad
+        buttonsContainerStackView.addArrangedSubview(yesButton)
+        if isIpad {
+            buttonsContainerStackView.addArrangedSubview(notNowButton)
+        }
+        contentView.addSubviews(cardTableView, buttonsContainerStackView)
+        view.addSubview(contentView)
+    }
+
+    func setupView() {
+        let headerHeight = UX.headerPreferredHeight
+        let estimatedCellHeight = cardTableView.estimatedRowHeight
+        let estimatedTableHeight = headerHeight + estimatedCellHeight + UX.distanceBetweenHeaderAndCells + UX.distanceBetweenButtonAndTable
+
+        tableViewHeightConstraint = cardTableView.heightAnchor.constraint(equalToConstant: estimatedTableHeight * getHeightScaleBasedOnFontSize())
+        tableViewHeightConstraint.priority = UILayoutPriority(999)
+
+        let contentViewWidth = UX.contentViewWidth > view.frame.width ? view.frame.width - UX.containerPadding : UX.contentViewWidth
+        contentWidthConstraint = contentView.widthAnchor.constraint(equalToConstant: contentViewWidth)
+        contentWidthConstraint.priority = UILayoutPriority(999)
+
+        NSLayoutConstraint.activate([
+            contentView.topAnchor.constraint(equalTo: view.topAnchor),
+            contentView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            contentView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+
+            cardTableView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: UX.distanceBetweenHeaderAndTop),
+            cardTableView.bottomAnchor.constraint(equalTo: buttonsContainerStackView.topAnchor, constant: 0),
+            cardTableView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: UX.tableMargin),
+            cardTableView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -UX.tableMargin),
+
+            buttonsContainerStackView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -UX.bottomSpacing),
+            buttonsContainerStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: UX.tableMargin),
+            buttonsContainerStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -UX.tableMargin),
+
+            yesButton.heightAnchor.constraint(equalToConstant: UX.yesButtonHeight),
+            notNowButton.heightAnchor.constraint(equalToConstant: UX.yesButtonHeight),
+            contentWidthConstraint,
+            tableViewHeightConstraint
+        ])
+    }
+
+    // MARK: View Transitions
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        print("traitCollectionDidChange")
+        setupView()
```

Comment:
If we need to adjust certain constraints when the trait collection changes, then we should do that instead of setting the constraints for a third time.